cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -79,6 +82,12 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
 
     update_discussions_settings_from_course_task.delay(course_key_str)
 
+@receiver(COURSE_CATALOG_INFO_CHANGED)
+def listen_for_course_catalog_info_changed(sender, signal, **kwargs):
+    print(f"kwargs: {kwargs}")
+    send_to_event_bus(COURSE_CATALOG_INFO_CHANGED, f"course-catalog-info-changed", 'catalog_info.course_key',
+                      {'catalog_info': kwargs['catalog_info']}, sync=True)
```

Comment:
From discussion: We need to figure out why sync=true was required during testing, so we can remove it.

Reply:
OK! Turns out the producer is getting deallocated at the end of the request. It [has a destructor](https://github.com/confluentinc/confluent-kafka-python/blob/d961259dfe25a65c23e2921be381da24eb24d253/src/confluent_kafka/src/Producer.c#L876) which [cleans up any threads](https://github.com/edenhill/librdkafka/blob/b47da0e4eec900d9cb091d2f524e957505dbb513/src/rdkafka.c#L1096) and other resources.

I experimented with event-bus-kafka and found that decorating `get_producer_for_signal` with `lru_cache` allows the event to send on this branch.

The good news is that creating an unbounded number of producers does not leak threads. The bad news is that even while iterating on this, we really can't get away with doing that, since it means producers die too quickly, and the events go with them. (I don't think `sync=True`, aka `flush()`, is a good idea in production.) I really think we need to switch to a single long-lived producer (or a small set of them).

Reply:
Updated https://github.com/openedx/event-bus-kafka/issues/16 with new information.